packages/block-renderer/src/hooks/use-pattern-inline-css.tsx

```diff
@@ -0,0 +1,53 @@
+import { shuffle } from '@automattic/js-utils';
+import { useMemo } from 'react';
+
+// Shuffle order of blog posts to avoid repetition in previews
+const shufflePostOrder = ( blogPostCount: number ) =>
+	shuffle( [ ...Array( blogPostCount ).keys() ] );
+
+// Memoize order per pattern to show same order in previews
+const inlineCssByPatternId: { [ key: string ]: string } = {};
+
+// Memoize last order
+let lastPostOrder: number[] = [];
+
+const usePatternInlineCss = ( patternId: string, patternHtml: string, isNewSite: boolean ) => {
+	return useMemo( () => {
+		const hasGrid = patternHtml?.includes( 'is-layout-grid' );
+		const blogPostCount = patternHtml?.match( /wp-block-post /g )?.length ?? 0;
+
+		// Only for patterns with a grid of posts in newly created sites
+		if ( ! isNewSite || ! hasGrid ) {
+			return undefined;
+		}
+
+		// Return memoized order
+		let inlineCss = inlineCssByPatternId[ patternId ] || '';
+		if ( inlineCss ) {
+			return inlineCss;
+		}
+
+		// Shuffle post order
+		let postOrder = shufflePostOrder( blogPostCount );
+
+		// Avoid repeating the last order
+		while ( lastPostOrder.toString() === postOrder.toString() ) {
```

Comment:
This would be an infinite loop if `blogPostCount` is `1` :) we should exit early from this hook when the pattern contains only 1 post.

Why do we need to avoid this scenario? Is it visually bad?

In any case, we can avoid the loop and just swap the last two elements when this happen.

Reply:
It avoids visual repetition between two consecutive patterns.

> we can avoid the loop and just swap the last two elements when this happen.

Let me try this 👍

Reply:
What if we just shuffle once more and remove the loop? I just pushed a new vesion.

Reply:
That line is only to prevent this case:

<img width="326" alt="Screenshot 2566-10-05 at 12 56 35" src="https://github.com/Automattic/wp-calypso/assets/1881481/1936f497-7b32-4391-950d-5f5b98cd52a7">